src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -422,8 +422,14 @@ private function validateAndDenormalize(string $currentClass, string $attribute,
             // if a value is meant to be a string, float, int or a boolean value from the serialized representation.
             // That's why we have to transform the values, if one of these non-string basic datatypes is expected.
             if (\is_string($data) && (XmlEncoder::FORMAT === $format || CsvEncoder::FORMAT === $format)) {
-                if ('' === $data && $type->isNullable() && \in_array($type->getBuiltinType(), [Type::BUILTIN_TYPE_BOOL, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT], true)) {
-                    return null;
+                if ('' === $data) {
+                    if (Type::BUILTIN_TYPE_ARRAY === $builtinType = $type->getBuiltinType()) {
+                        return [];
+                    }
+
+                    if ($type->isNullable() && \in_array($builtinType, [Type::BUILTIN_TYPE_BOOL, Type::BUILTIN_TYPE_INT, Type::BUILTIN_TYPE_FLOAT], true)) {
+                        return null;
+                    }
                 }
 
                 switch ($type->getBuiltinType()) {
```

Comment:
We can even do this then:

```suggestion
                switch ($builtinType ?? $type->getBuiltinType()) {
```

Reply:
Saw you also added it in #43240 for 5.4, just pushed it for 5.3. Thanks for the approval!